go-selinux/selinux_linux.go

```diff
@@ -697,17 +696,21 @@ func socketLabel() (string, error) {
 
 // peerLabel retrieves the label of the client on the other side of a socket
 func peerLabel(fd uintptr) (string, error) {
-	return unix.GetsockoptString(int(fd), unix.SOL_SOCKET, unix.SO_PEERSEC)
+	label, err := unix.GetsockoptString(int(fd), unix.SOL_SOCKET, unix.SO_PEERSEC)
+	if err != nil {
+		return "", &os.PathError{Op: "getsockopt", Path: "fd " + strconv.Itoa(int(fd)), Err: err}
+	}
+	return label, nil
 }
 
 // setKeyLabel takes a process label and tells the kernel to assign the
 // label to the next kernel keyring that gets created
 func setKeyLabel(label string) error {
 	err := writeCon("/proc/self/attr/keycreate", label)
-	if os.IsNotExist(errors.Cause(err)) {
+	if os.IsNotExist(err) {
 		return nil
 	}
-	if label == "" && os.IsPermission(errors.Cause(err)) {
+	if label == "" && os.IsPermission(err) {
```

Comment:
Should we use `errors.Is(err, os.ErrPermission)`?

https://golang.org/pkg/os/#IsPermission

> This function predates errors.Is. It only supports errors returned by the os package. New code should use errors.Is(err, os.ErrPermission).

Reply:
Wow, I did not know that the official recommendation is to use `errors.Is` instead of `os.IsXxx`. Will fix.

Reply:
fixed

Reply:
Actually fixed 🤞🏻